javascript/modules/client/src/client.ts

```diff
@@ -546,27 +549,48 @@ export class Client extends ClientCore implements IClient {
       override.value = amount;
     }
 
-    // TODO: If the approved ERC20 amount is not enough,
-    // estimate the cose of increasing the allowance
-
     return daoInstance.estimateGas
       .deposit(tokenAddress, amount, reference, override)
       .then((gasLimit) => {
         return this.web3.getApproximateGasFee(gasLimit.toBigInt());
       });
   }
 
-  _estimateUpdateAllowance(_params: IDepositParams) {
+  async _estimateUpdateAllowance(params: IDepositParams) {
     const signer = this.web3.getConnectedSigner();
     if (!signer) {
-      throw new Error("A signer is needed");
+      throw new NoSignerError();
     } else if (!signer.provider) {
-      throw new Error("A web3 provider is needed");
+      throw new NoProviderError();
+    } else if (!params.tokenAddress) {
+      throw new NoTokenAddress();
     }
-    // TODO: remove this
-    return Promise.resolve(
-      this.web3.getApproximateGasFee(Random.getBigInt(BigInt(1500))),
+    // resolve ens ?
+    // let daoAddreess = params.daoAddress;
+    // if (!isAddress(daoAddreess)) {
+    //   await this.web3.ensureOnline();
+    //   const provider = this.web3.getProvider();
+    //   if (!provider) {
+    //     throw new NoProviderError();
+    //   }
+    //   const resolvedAddress = await provider.resolveName(daoAddreess);
+    //   if (!resolvedAddress) {
+    //     throw new InvalidAddressOrEnsError();
+    //   }
+    //   daoAddreess = resolvedAddress;
+    // }
```

Comment:
Why this comment? Can't we refactor this somewhere else and leave clean code here?

Reply:
This comment was more of a doubt if we wanted ens resolution in this function, if we don't want it I'll remove the comment and the `async`, if we want to resolve it, I'll just uncomment

Reply:
ok, but then why don't we uncomment this part and have the feature?

Reply:
I wasn't sure that we wanted this feature here because we don't have it in `deposit` or `estimation.deposit` but I'll uncomment it and add a `TODO` to update the `deposit` and `estimation.deposit` later

Reply:
In general, we should be able to handle ENS names everywhere, as the intended way to interact with DAO's will be with URL's where the parameter is a domain name, not a hex string (ideally)